airflow/jobs/scheduler_job_runner.py

```diff
@@ -1298,7 +1298,7 @@ def _should_update_dag_next_dagruns(self, dag: DAG, dag_model: DagModel, total_a
 
     def _start_queued_dagruns(self, session: Session) -> None:
         """Find DagRuns in queued state and decide moving them to running state."""
-        dag_runs: Collection[DagRun] = self._get_next_dagruns_to_examine(DagRunState.QUEUED, session)
+        dag_runs: Collection[DagRun] = self._get_next_dagruns_to_examine(DagRunState.QUEUED, session).all()
```

Comment:
Probably worthwhile to add a comment here so someone in the future don’t try to be smart and remove the `all()` call.